app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/auth/oauth/OAuthState.java

```diff
@@ -26,16 +26,30 @@
     private volatile State state;
 
     private static final class State {
-        private final String accessToken;
+        private final String _accessToken;
 
-        private final long accessTokenExpiresAt;
+        private final long _accessTokenExpiresAt;
 
-        private final String refreshToken;
+        private final String _refreshToken;
+
+        public synchronized String getAccessToken() {
+            return _accessToken;
+        }
+
+        public synchronized long getAccessTokenExpiresAt() {
+            return _accessTokenExpiresAt;
+        }
+
+        public synchronized String getRefreshToken() {
+            return _refreshToken;
+        }
 
         private State(final String accessToken, final long accessTokenExpiresAt, final String refreshToken) {
-            this.accessToken = accessToken;
-            this.refreshToken = refreshToken;
-            this.accessTokenExpiresAt = accessTokenExpiresAt;
+            synchronized (this){
```

Comment:
This should not be required, why lock here?

Reply:
I was trying to make the "volatile" useful. When applied to objects, volatile does not warrantee multithreading readings.

Reply:
If the "volatile" label is not important in this context, then we can revert this commit.

Reply:
I think in this case `volatile` is is used correctly, it can be replaced with `AtomicReference` as an alternative that might be less confusing. Adding synchronization, especially in the constructor doesn't make much sense. Going through setters for `private final` field also doesn't make much sense, in fact in this case it might lead to `StackOverflowErrors`.

Reply:
Ok, then I will remove this commit from the list.